app/controllers/sprangular/carts_controller.rb

```diff
@@ -11,13 +11,21 @@ def show
 
   # Adds a new item to the order (creating a new order if none already exists)
   def add_variant
-    populator = Spree::OrderPopulator.new(current_order(create_order_if_necessary: true), current_currency)
-    if populator.populate(params[:variant_id], params[:quantity])
-      current_order.ensure_updated_shipments
-      @order = current_order
-      render 'spree/api/orders/show'
+    order = current_order(create_order_if_necessary: true)
+    variant  = Spree::Variant.find(params[:variant_id])
+    quantity = params[:quantity].to_i
+
+    if quantity.between?(1, 2_147_483_647)
+      begin
+        order.contents.add(variant, quantity)
+        order.ensure_updated_shipments
+        @order = order.reload
+        render 'spree/api/orders/show'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.